src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -861,10 +862,16 @@ impl AccessChecker for PrivilegeAccess {
                 self.validate_table_access(&plan.catalog, &plan.database, &plan.table, vec![UserPrivilegeType::Update], false).await?;
             }
             Plan::CreateView(plan) => {
-                self.validate_db_access(&plan.catalog, &plan.database, vec![UserPrivilegeType::Create], false).await?
+                self.validate_db_access(&plan.catalog, &plan.database, vec![UserPrivilegeType::Create], false).await?;
```

Comment:
I think View just needs Select privilege for inner table even in Create Or Alter.

Reply:
If so, we no need to add ownership cover view which will be easier . But I think this is a product form. 
So cc @flaneur2020 how about you?

If a user can query the inner table in view, it can create the view.

But I think alter and drop view need privilege check about alter and drop privilege on db level.

Reply:
> I think View just needs Select privilege for inner table even in Create Or Alter.

About alter view, if user a modify the v. After some days User b also modify the view v to adopt b's business.

It maybe destroys a.

So I think alter and drop need to check privileges.

Reply:
Modify in newest commit.

Reply:
found the related semantics from snowflake:

> When you create a view and then grant privileges on that view to a role, the role can use the view even if the role does not have privileges on the underlying table(s) that the view accesses. This means that you can create a view to give a role access to only a subset of a table. For example, you can create a view that accesses medical billing information but not medical diagnosis information in the same table. Then you can grant privileges on that view to the “accountant” role so that the accountants can look at the billing information without seeing the patient’s diagnosis.

it seems that if you have access to a view, even you can not access the underlying tables, you can still access the underlying table data by the view.

Reply:
> If a user can query the inner table in view, it can create the view.
>
> But I think alter and drop view need privilege check about alter and drop privilege on db level.

agree, if we hope using view as a "public interface" for others to access the unaccessible table data, we need carefully check the permissions when we create/alter the view, ensuring the one whom created the view, have all the SELECT permissions to the underlying tables.

Reply:
> found a the related docs from snowflake:
> 
> > When you create a view and then grant privileges on that view to a role, the role can use the view even if the role does not have privileges on the underlying table(s) that the view accesses. This means that you can create a view to give a role access to only a subset of a table. For example, you can create a view that accesses medical billing information but not medical diagnosis information in the same table. Then you can grant privileges on that view to the “accountant” role so that the accountants can look at the billing information without seeing the patient’s diagnosis.
> 
> it seems that if you have access to a view, even you can not access the underlying tables, you can still access the underlying table data by the view.

Yes. That's why we need view. 

I also think do not need all inner tables. e.g.: v_tt is view of table `tt`, v_v_tt is view of view `v_tt`

-- user : a
create table tt(id int);

insert into tt values(1);

create view v_tt as select * from tt;

-- user : a

create view v_v_tt as select * from v_tt;
ERROR 1105 (HY000): PermissionDenied. Code: 1063, Text = Permission denied: privilege [Select] is required on 'default'.'default'.'v_tt' for user 'a'@'%' with roles [public].

-- user: root grant select on view to user a
grant select on default.v_tt to a;
grant select on default.v_v_tt to a;

--user : a
-- even user can not access inner table in v_t, but user a can access view v_t, that's enough
select * from v_v_tt;
+------+
| id   |
+------+
|    1 |
+------+

Reply:
> I also think do not need all inner tables. e.g.: v_tt is view of table tt, v_v_tt is view of view v_tt

oh yes, we need ensure the creator can access the inner tables & the inner views of the SQL which creating the view, but we do not need to further check the permission of inner tables of the inner views.